src/nms-api/IConnection.cs

```diff
@@ -95,16 +96,22 @@ public interface IConnection : IDisposable, IStartable, IStoppable
         /// </summary>
         ISession CreateSession();
 
+        Task<ISession> CreateSessionAsync();
```

Comment:
It would be wise to have a CancellationToken as a parameter to all `*Async` methods.